### DISCUSSION TITLE
Azure DevOps - with ESLZ, first run 'AzOps - Push' pipeline - next first Pull then Push's EVERYTHING, times out (60 minutes)

### DISCUSSION THREAD

**Discussion by mikewo-dc** [Azure/AzOps#667] [Q&A] [answered]
I have encountered this a few times now - enough to wonder what is happening - Can anyone help explain and/or validate the below? 

The scenario is:

I have an Enterprise Scale Landing Zone with management group structure, policy definitions and numerous policies assigned directly or via the ESLZ initiatives.

I deploy AzOps-Accelerator in Azure DevOps - as per [https://github.com/azure/azops/wiki/azure-pipelines](https://github.com/azure/azops/wiki/azure-pipelines)

I run the "Azure - Push" pipeline first - as per instruction "trigger the first push, which will in turn trigger the first pull to fetch the existing Azure environment"

*** The resulting abovementioned "first pull" triggers ANOTHER Push, and this Push goes on forever, attempting to redeploy everything...

... until the Azure DevOps hosted agent reaches the run time limit of 60 minutes. The pipeline job errors out on the Deploy step.

Is this expected behaviour? And if so why does the first Pull trigger a Push of everything?

Appreciate any help. I am happy to try reproduce and log an issue if this is not expected.

**Comment by daltondhcp** [comment, marked as answer]
Hi @mikewo-dc,

This is currently the default behavior with ADO. With GitHub Actions  it is not automatically triggered after the initial pull. If you want to change the default behavior in ADO, you can find a solution in the following thread. 
https://github.com/Azure/AzOps/discussions/648#discussioncomment-2857924

We are thinking about adding that specific condition to the ADO starter pipeline but haven't come to that yet. 

Hope this helps. 

/Johan

**Reply by mikewo-dc** [reply]
Legend - thank you heaps. I've added the condition to my push.yml as per the link, thanks for helping out a noob.
